ui/src/app/base/constants.ts

```diff
@@ -1,3 +1,10 @@
+export const ACTION_STATUS = {
+  failed: "failed",
+  idle: "idle",
+  processing: "processing",
+  successful: "successful",
```

Comment:
I'd suggest to replace this with a slightly shorter status strings format that's used in many different libraries (e.g. https://github.com/tannerlinsley/react-query/blob/f2137dc4e4553256c4ebc1891b548fe35efe9231/src/core/types.ts#L250).

The convention below is widely adopted. Slight semantic difference between failed and error, or processing and loading doesn't seem to matter in this case - AFAIK the conceptual categories that we're referring to are the same.

```suggestion
  idle: "idle",
  loading: "loading",
  error: "error",
  success: "success",
```

Reply:
Good suggestion, thanks! I guess I was coming up with terms that describe what happened (this thing failed, this thing was successful) but it might be more appropriate to describe the state declaratively instead (this thing is in error state, this thing is in success state).